Test/Unit/Model/Api/ShippingMethodsTest.php

```diff
@@ -0,0 +1,322 @@
+use Magento\Framework\Webapi\Rest\Response;
+use Bolt\Boltpay\Helper\Config as ConfigHelper;
+use Magento\Framework\Webapi\Rest\Request;
+use Magento\Framework\App\CacheInterface;
+use Magento\Framework\Pricing\Helper\Data as PriceHelper;
+use Bolt\Boltpay\Model\ErrorResponse as BoltErrorResponse;
+use Bolt\Boltpay\Helper\Session as SessionHelper;
+
+/**
+ * Class ShippingMethodsTest
+ *
+ * @package Bolt\Boltpay\Test\Unit\Model\Api
+ */
+class ShippingMethodsTest extends TestCase
+{
+    /**
+     * @var BoltShippingMethods
+     */
+    private $currentMock;
+
+    private $hookHelper;
+
+    private $regionModel;
+
+    private $factoryShippingOptionsMock;
+
+    private $shippingTaxInterfaceFactory;
+
+    private $shippingOptionInterfaceFactory;
+
+    private $totalsCollector;
+
+    private $converter;
+
+    private $logHelper;
+
+    private $errorResponse;
+
+    private $response;
+
+    private $configHelper;
+
+    private $request;
+
+    private $cache;
+
+    private $priceHelper;
+
+    private $sessionHelper;
+
+    private $bugsnag;
+
+    /**
+     * @inheritdoc
+     */
+    public function setUp()
+    {
+        $this->factoryShippingOptionsMock = $this->getMockBuilder(ShippingOptionsInterfaceFactory::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['create'])
+            ->getMock();
+
+        $this->shippingTaxInterfaceFactory = $this->getMockBuilder(ShippingTaxInterfaceFactory::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['create'])
+            ->getMock();
+
+        $this->shippingOptionInterfaceFactory = $this->getMockBuilder(ShippingOptionInterfaceFactory::class)
+            ->setMethods(['create'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->hookHelper = $this->createMock(HookHelper::class);
+        $this->regionModel = $this->createMock(RegionModel::class);
+        $this->totalsCollector = $this->createMock(TotalsCollector::class);
+        $this->converter = $this->createMock(ShippingMethodConverter::class);
+        $this->logHelper = $this->createMock(LogHelper::class);
+        $this->errorResponse = $this->createMock(BoltErrorResponse::class);
+        $this->response = $this->createMock(Response::class);
+        $this->configHelper = $this->createMock(ConfigHelper::class);
+        $this->request = $this->createMock(Request::class);
+        $this->cache = $this->createMock(CacheInterface::class);
+        $this->priceHelper = $this->createMock(PriceHelper::class);
+        $this->sessionHelper = $this->createMock(SessionHelper::class);
+
+        $this->bugsnag = $this->getMockBuilder(Bugsnag::class)
+            ->setMethods(['notifyException'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->bugsnag->method('notifyException')
+            ->willReturnSelf();
+    }
+
+    public function testGetShippingMethodsIfQuoteEmpty()
```

Comment:
it's better if test can also say expected outcome - eg., `testGetShippingMethodsRetrunsErrorIfQuoteEmpty`